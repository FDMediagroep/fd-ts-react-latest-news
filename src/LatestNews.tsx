import React, { PureComponent } from "react";
import Card, { CardStyle, CardTypes } from "@fdmg/fd-card";
import TypoGraphy, { getAllTextStyles } from "@fdmg/fd-typography";
import { createGlobalStyle } from "styled-components";

interface NewsItem {
    uuid: string;
    dateTime: string;
    isRead?: boolean;
    target?: string;
    title: string;
    url: string;
}

export interface Props {
    cardStyle: CardTypes;
    className?: string;
    link: string;
    linkText?: string;
    news: NewsItem[];
    title?: string;
}

export default class LatestNews extends PureComponent<Props, any> {

    render() {
        return (
            <>
                <GlobalStyle/>
                <Card cardStyle={this.props.cardStyle} className={`fd-card-latest-news${this.props.className ? ` ${this.props.className}` : ''}`}>
                    <TypoGraphy className="h-latest-news" textStyle='card-h'><h3>{this.props.title ? this.props.title : 'Laatste nieuws'}</h3></TypoGraphy>
                    <nav>
                        {
                            this.props.news.map((newsItem) => {
                                return (
                                    <a key={newsItem.uuid} href={newsItem.url} target={newsItem.target}>
                                        <time>{newsItem.dateTime}</time>
                                        <span className={newsItem.isRead ? 'is-read' : undefined}>{newsItem.title}</span>
                                    </a>
                                );
                            })
                        }
                    </nav>
                    <a href={this.props.link}>{this.props.linkText ? this.props.linkText : 'Lees al het laatste nieuws'} <i className="icon-chevron-right"/></a>
                </Card>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-card-latest-news {
    .h-latest-news,
    > a,
    nav > a {
        padding: 12px 10px;
        @media only screen and (min-width: 641px) {
            padding: 12px;
        }
        @media only screen and (min-width: 861px) {
            padding: 12px 15px;
        }
    }

    &.persoonlijk .h-latest-news {
        color: #e57e30;
    }

    > a,
    nav > a {
        display: block;
        font-family: 'ProximaNovaRegular', sans-serif;
        border-top: 1px solid rgba(0,0,0,0.1);
        line-height: 1.2em;
        color: #191919;
        .is-read {
            opacity: 0.6;
        }
        &:hover {
            color: #49a4a2;
            .icon-chevron-right {
                color: #49a4a2;
            }
        }
        text-decoration: none;
        font-size: 16px;
        @media only screen and (min-width: 641px) {
            font-size: 17px;
        }
    }

    .icon-chevron-right {
        float: right;
        color: #677381;
        font-size: 0.8125rem;
        margin-top: 4px;
    }

    time {
        color: #677381;
        display: block;
        font-size: 13px;
        @media only screen and (min-width: 641px) {
            font-size: 14px;
        }
    }
}
`;

export const LatestNewsStyle = createGlobalStyle`
    ${(CardStyle as any).globalStyle.rules}
    ${getAllTextStyles(['card-h']).globalStyle.rules}
    ${(GlobalStyle as any).globalStyle.rules}
`;
