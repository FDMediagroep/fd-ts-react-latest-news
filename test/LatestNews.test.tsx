import TestRenderer from 'react-test-renderer';
import LatestNews from '../src/LatestNews';
import React from 'react';

describe('LatestNews', () => {
    test('renders correctly', () => {
        let latestNews = TestRenderer.create(<LatestNews cardStyle="default" link='/laatste-nieuws' news={[{uuid: '1', dateTime: '15:29', title: 'title van nieuwsbericht', url: 'https://fd.nl'}]}/>);
        expect(latestNews.toJSON()).toMatchSnapshot();
        latestNews = TestRenderer.create(<LatestNews className="class-name" cardStyle="article" linkText="Ga naar laatste nieuws" title="Laatste nieuws blokje" link='/laatste-nieuws' news={[{uuid: '2', dateTime: '15:29', title: 'title van nieuwsbericht', url: 'https://fd.nl', target: "_blank"}]}/>);
        expect(latestNews.toJSON()).toMatchSnapshot();
    });

    test('renders is-read correctly', () => {
        let latestNews = TestRenderer.create(<LatestNews link='/laatste-nieuws' cardStyle="persoonlijk" news={[{uuid: '1', dateTime: '15:29', isRead: true, title: 'title van nieuwsbericht', url: 'https://fd.nl'}]}/>);
        expect(latestNews.toJSON()).toMatchSnapshot();
    });
});
