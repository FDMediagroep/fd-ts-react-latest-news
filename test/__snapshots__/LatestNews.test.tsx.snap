// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LatestNews renders correctly 1`] = `
<div
  className="fd-card fd-card-latest-news"
>
  <h3
    className="fd-typography card-h h-latest-news"
    textStyle="card-h"
  >
    Laatste nieuws
  </h3>
  <nav>
    <a
      href="https://fd.nl"
    >
      <time>
        15:29
      </time>
      <span>
        title van nieuwsbericht
      </span>
    </a>
  </nav>
  <a
    href="/laatste-nieuws"
  >
    Lees al het laatste nieuws
     
    <i
      className="icon-chevron-right"
    />
  </a>
</div>
`;

exports[`LatestNews renders correctly 2`] = `
<div
  className="fd-card fd-card-latest-news class-name"
>
  <h3
    className="fd-typography card-h h-latest-news"
    textStyle="card-h"
  >
    Laatste nieuws blokje
  </h3>
  <nav>
    <a
      href="https://fd.nl"
      target="_blank"
    >
      <time>
        15:29
      </time>
      <span>
        title van nieuwsbericht
      </span>
    </a>
  </nav>
  <a
    href="/laatste-nieuws"
  >
    Ga naar laatste nieuws
     
    <i
      className="icon-chevron-right"
    />
  </a>
</div>
`;

exports[`LatestNews renders is-read correctly 1`] = `
<div
  className="fd-card fd-card-latest-news"
>
  <h3
    className="fd-typography card-h h-latest-news"
    textStyle="card-h"
  >
    Laatste nieuws
  </h3>
  <nav>
    <a
      href="https://fd.nl"
    >
      <time>
        15:29
      </time>
      <span
        className="is-read"
      >
        title van nieuwsbericht
      </span>
    </a>
  </nav>
  <a
    href="/laatste-nieuws"
  >
    Lees al het laatste nieuws
     
    <i
      className="icon-chevron-right"
    />
  </a>
</div>
`;
